x-pack/plugins/fleet/server/routes/message_signing_service/handlers.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { TypeOf } from '@kbn/config-schema';
+
+import type { FleetRequestHandler } from '../../types';
+
+import { defaultFleetErrorHandler, AgentPolicyNotFoundError } from '../../errors';
+import { appContextService } from '../../services';
+import type { RotateKeyPairSchema } from '../../types/rest_spec/message_signing_service';
+
+export const rotateKeyPairHandler: FleetRequestHandler<
+  TypeOf<typeof RotateKeyPairSchema.params>,
+  undefined,
+  undefined
+> = async (_, request, response) => {
+  if (!request.params.acknowledge) {
```

Comment:
for some reason the param is always `false` when I'm testing with `POST api/fleet/messageSigningService/rotateKeyPair?acknowledge=true`. Maybe I missed something. @dasansol92 would you mind testing it out?